NitroxClient/Communication/Packets/Processors/PrecursorDoorToggleProcessor.cs

```diff
@@ -0,0 +1,31 @@
+﻿using NitroxClient.Communication.Abstract;
+using NitroxClient.Communication.Packets.Processors.Abstract;
+using NitroxClient.MonoBehaviours;
+using UnityEngine;
+
+namespace NitroxClient.Communication.Packets.Processors
+{
+    class PrecursorDoorToggleProcessor : ClientPacketProcessor<NitroxModel.Packets.PrecursorDoorwayToggle>
+    {
+        private readonly IPacketSender packetSender;
+
+        public PrecursorDoorToggleProcessor(IPacketSender packetSender)
+        {
+            this.packetSender = packetSender;
+        }
+
+        public override void Process(NitroxModel.Packets.PrecursorDoorwayToggle packet)
+        {
+            GameObject gameObject = NitroxEntity.RequireObjectFrom(packet.Id);
+            PrecursorDoorway precursorDoorway = gameObject.GetComponent<PrecursorDoorway>();
+
+            if (packet.ToggleDoorway != precursorDoorway.isOpen)
+            {
+                using (packetSender.Suppress<NitroxModel.Packets.PrecursorDoorwayToggle>())
+                {
+                    precursorDoorway.ToggleDoor(packet.ToggleDoorway);
```

Comment:
Will this work if the other player is not nearby?  If this entity is not in the global root then it may be out of sync.

Reply:
You are correct it doesn't work, also when another player also comes close it actually puts the forcefield back. Any ideas on this?